bilge-impl/src/bitsize.rs

```diff
@@ -143,17 +149,21 @@ fn generate_struct(item: &ItemStruct, declared_bitsize: u8) -> TokenStream {
         }
     };
 
+    let (impl_generics, ty_generics, where_clause) = generics.split_for_impl();
+
     quote! {
-        #vis struct #ident #fields_def
-
-        // constness: when we get const blocks evaluated at compile time, add a const computed_bitsize
-        const _: () = assert!(
-            (#computed_bitsize) == (#declared_bitsize),
-            concat!("struct size and declared bit size differ: ",
-            // stringify!(#computed_bitsize),
-            " != ",
-            stringify!(#declared_bitsize))
-        );
+        #vis struct #ident #generics #fields_def
```

Comment:
this doesn't account for structs with where clauses:
```rust
#[bitsize(2)]
struct IsGenericWithWhere<T>(Generic<T>) where T: Copy + Clone;
```
this struct would lose the `where T: Copy + Clone`.

I think code expansion here would need to account for (among other things), the difference between tuple structs and structs with named fields. the former has the where clause before `#fields_def`, the latter has it after.

Reply:
This should be fixed now, but I haven't written a test for it yet. should be some kind of "shouldn't compile" test.

Reply:
A test has been written.